### PR TITLE
[xtext generator] changed bundle activator naming computation to be grammar independent

### PR DESCRIPTION
**Pull request by sailingKieler**
* moved activator name computation to 'XtextGeneratorNaming',
* fixed basically two bugs in 'XtextGenerator' concerning the setting of the bundle activator,
* added dedicated getter & setter in 'MergableManifest' updating the 'modified' flag,
* updated usage of the eclipse plugin activator in 'JUnit4Fragment2'

Signed-off-by: Christian Schneider <christian.schneider@itemis.de>

### CODE REVIEW

**Comment by sailingKieler** [comment]
Jenkins test

File: plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorNaming.xtend [hunk 1]
```diff
@@ -69,8 +74,25 @@ class XtextGeneratorNaming {
...
+		// start determining the activator's name by stripping the common ".ui" suffix
+		var activatorName = pluginName.replaceAll('\\.ui$', '')
+		
+		// build the activator's name based on the last segment of the stripped FQN
+		//  the call of 'new TypeReference(...)' below checks for invalid characters
+		activatorName = activatorName.substring(activatorName.lastIndexOf('.') + 1).toFirstUpper
+		
+		if (activatorName != pluginName) {
+			// ... pluginName ends with ".ui" as by default, so
+			activatorName += 'UIActivator'
```
Comment (svenefftinge) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorNaming.xtend]: Why a UI prefix here?

Comment (sailingKieler) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorNaming.xtend]: To indicate that it's not the language defining plug-in's activator but that of the plug-in contributing the UI stuff.
But it's not that important.

**Comment by svenefftinge** [comment]
+1

**Comment by sailingKieler** [comment]
Jenkins test

**Comment by sailingKieler** [comment]
Please don't merge yet, need to check some things.

**Comment by sailingKieler** [comment]
Jenkins test

**Pull request merged**